### ISSUE TITLE
feat: add no-trailing-punctuation rule (MD026)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#189]
## Overview

Implement the `no-trailing-punctuation` rule (equivalent to markdownlint MD026 / remark-lint `no-heading-punctuation`).

Closes part of #76.

## Problem

Headings ending with punctuation such as `.`, `,`, `;`, `:`, or `!` are a common mistake that reduces document quality. Most style guides discourage trailing punctuation in titles and headings.

**Example of the problem:**
```markdown
# Introduction.
## Prerequisites:
### Warning!
```

## Proposed Rule Key

`no-trailing-punctuation`

## Detection Logic

| Input | Violation? | Reason |
|---|---|---|
| `# Heading.` | Yes | Ends with `.` |
| `## Section,` | Yes | Ends with `,` |
| `### Title!` | Yes | Ends with `!` |
| `#### Why?` | No | `?` is not in the default set |
| `# Heading` | No | No trailing punctuation |
| `## Config: Options` | No | `:` is in the middle, not trailing |
| `### Example (Part 1)` | No | Ends with `)` which is not in the default set |
| `# [Link](url).` | Yes | Trailing `.` after link |

Setext-style headings are also checked:

| Input | Violation? |
|---|---|
| `Heading.\n========` | Yes |
| `Heading\n========` | No |

Content inside fenced code blocks must be ignored.

## Fix

Remove the trailing punctuation character from the heading text.

## Config

```json
"no-trailing-punctuation": {
  "enabled": true,
  "punctuation": ".,;:!"
}
```

- `punctuation`: string of characters that must not appear at the end of a heading (default: `.,;:!`)
- `?` is intentionally excluded from the default — question-style headings (e.g., "Why does this matter?") are common and valid. Add `?` to `punctuation` to opt in.

## References

- [markdownlint MD026](https://github.com/DavidAnson/markdownlint/blob/main/doc/md026.md)
- [remark-lint no-heading-punctuation](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-heading-punctuation)

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create
- `internal/rule/no_trailing_punctuation.go` — `CheckNoTrailingPunctuation` function
- `internal/rule/no_trailing_punctuation_test.go` — unit tests (table-driven)
- `e2e/fixtures/no_trailing_punctuation_valid.md` — valid fixture
- `e2e/fixtures/no_trailing_punctuation_violation.md` — violation fixture
- `e2e/fixtures/config-no-trailing-punctuation.json` — e2e config

### Files to modify
- `internal/config/config.go` — register `no-trailing-punctuation` in `DefaultConfigJSON` and `Default()`
- `internal/linter/linter.go` — call `CheckNoTrailingPunctuation` in `collectErrors()`
- `e2e/e2e_test.go` — add e2e test cases
- `cmd/root_bench_test.go` — add violation-free heading content to `generateComplexMarkdown`

### Checklist
- [ ] Implement `CheckNoTrailingPunctuation(filename string, lines []string, cfg Config, offset int)` in rule package
  - Strip trailing whitespace from the heading text before checking
  - Detect ATX headings (`# Heading`) — extract text after the `#` prefix
  - Detect setext headings (underline with `===` or `---`) — check the line above the underline
  - Check if the last character of the heading text is in the configured `punctuation` string
  - Skip fenced code blocks (backtick and tilde)
  - Reuse `openingFenceMarker`, `IsClosingFence` utilities
- [ ] Write unit tests covering:
  - Each default punctuation character (`. , ; : !`) triggers a violation
  - `?` does not trigger a violation with the default config
  - `?` triggers a violation when added to the `punctuation` config
  - Heading with no trailing punctuation is valid
  - Punctuation in the middle of a heading (e.g., `## Config: Options`) is valid
  - Setext headings with and without trailing punctuation
  - Headings inside fenced code blocks (should be ignored)
  - Custom `punctuation` config value
  - Offset handling for frontmatter
  - Multiple violations across multiple headings
- [ ] Register rule in config (`DefaultConfigJSON` + `Default()`) with `enabled: true` and `punctuation: ".,;:!"`
- [ ] Integrate in linter `collectErrors()`
- [ ] Add e2e test fixtures and test cases
- [ ] Add violation-free headings to `generateComplexMarkdown` in `cmd/root_bench_test.go` and verify `TestBenchmarkContentIsViolationFree` passes
- [ ] Run `make test-all`
